boat-maven-plugin/src/main/java/com/backbase/oss/boat/RadioMojo.java

```diff
@@ -0,0 +1,171 @@
+package com.backbase.oss.boat;
+
+import com.backbase.oss.boat.bay.client.ApiClient;
+import com.backbase.oss.boat.bay.client.api.BoatMavenPluginApi;
+import com.backbase.oss.boat.bay.client.model.BoatLintReport;
+import com.backbase.oss.boat.bay.client.model.UploadRequestBody;
+import com.backbase.oss.boat.bay.client.model.UploadSpec;
+import com.backbase.oss.boat.loader.OpenAPILoader;
+import com.backbase.oss.boat.loader.OpenAPILoaderException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationConfig;
+import feign.auth.BasicAuthRequestInterceptor;
+import io.swagger.v3.oas.models.OpenAPI;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "radio", requiresDependencyResolution = ResolutionScope.RUNTIME, threadSafe = true)
+@Slf4j
+@Getter
+@Setter
+public class RadioMojo extends AbstractMojo {
+
+    @Parameter(property = "groupId", defaultValue = "${project.groupId}", required = true)
+    private String groupId;
+
+    @Parameter(property = "artifactId", defaultValue = "${project.artifactId}", required = true)
+    private String artifactId;
+
+    @Parameter(property = "version", defaultValue = "${project.version}", required = true)
```

Comment:
"The property parameter can be used to allow configuration of the mojo parameter from the command line by referencing a system property that the user sets via the -D option.", hence it is normal to namespace them. e.g. `boat.radio` I would expect there could be other maven plugins/mojos with properties called things like version, or even system properties like `username`. The output parameter can then be just output/boat.radio.output etc.

Reply:
boat-bay radio is to be executed on spec projects as created by the ssdk archetype. We should inherit the GAV from the project. Why make these configurable?

Reply:
GAV?

Reply:
I was a bit confused about where the plugin would be used. We usually have:

1. specs 
2. api (built from specs)
3. BOM (reference above)

 At the moment, a project can only be registered via BOM (bom are only cal_ver). Can the plugin be used from api/spec level? Or it can only be used from the BOM level only. In that case, I guess I have unzip and parse through all files.

Reply:
@bartveenstra Should we add unzip logic incase the plugin is used from bom.
Or are we planning to add project creation in boatbay other than bom too?